src/document/DocumentAnnotator.ts

```diff
@@ -1,21 +1,46 @@
+import debounce from 'lodash/debounce';
 import BaseAnnotator, { Options } from '../common/BaseAnnotator';
 import BaseManager from '../common/BaseManager';
 import { centerRegion, isRegion, RegionManager } from '../region';
 import { Event } from '../@types';
 import { getAnnotation } from '../store/annotations';
-import { Mode } from '../store';
+import { getSelectionItem } from './docUtil';
+import { Mode, setSelectionAction } from '../store';
 import { scrollToLocation } from '../utils/scroll';
 import './DocumentAnnotator.scss';
 
+const TEXT_LAYER_ENHANCEMENT = 300;
+
 export default class DocumentAnnotator extends BaseAnnotator {
     annotatedEl?: HTMLElement;
 
     managers: Map<number, Set<BaseManager>> = new Map();
 
+    selectionChangeTimer: number | null = null;
```

Comment:
Is `null` meaningfully different from `undefined` here? If not, can we use the latter value, instead?